tests/integration/api/materialization_pulsar_test.go

```diff
@@ -0,0 +1,216 @@
+package api_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	v1alpha "github.com/kaskada-ai/kaskada/gen/proto/go/kaskada/kaskada/v1alpha"
+	helpers "github.com/kaskada-ai/kaskada/tests/integration/shared/helpers"
+	. "github.com/kaskada-ai/kaskada/tests/integration/shared/matchers"
+)
+
+type testSchema struct {
+	Key       string `json:"key"`
+	MaxAmount int    `json:"max_amount"`
+	MinAmount int    `json:"min_amount"`
+}
+
+var _ = Describe("Materialization with Pulsar upload", Ordered, Label("pulsar"), func() {
```

Comment:
Depending on if we fix the integration tests (@therapon is working on this), we may skip this so as to not block merge to main